Lib/multiprocessing/managers.py

```diff
@@ -671,13 +673,13 @@ def _finalize_manager(process, address, authkey, state, _Client):
             except Exception:
                 pass
 
-            process.join(timeout=1.0)
+            process.join(timeout=shutdown_timeout)
             if process.is_alive():
                 util.info('manager still alive')
                 if hasattr(process, 'terminate'):
                     util.info('trying to `terminate()` manager process')
                     process.terminate()
-                    process.join(timeout=0.1)
+                    process.join(timeout=shutdown_timeout)
                     if process.is_alive():
                         util.info('manager still alive after terminate')
```

Comment:
SIGTERM can be caught or ignored (or perhaps delayed?), so it would make sense to forcefully kill the process here:
```suggestion
                        util.info('manager still alive after terminate')
                        process.kill()
                        process.join()
```

Reply:
cc @gps for opinion

Reply:
subprocess.call() kills the process if wait() raises an exception like KeyboardInterrupt. The rationale for this function is that the caller does not have access to the process, so Python tries to be a good citizen and kill the process.

Here, it's a similar situation: _finalize_manager() can be called automatically by the multiprocessing "atexit" mecanism based on reference counting or the Python GC: when the manager is gone, and the developer was too lazy to cleanly shutdown the process. So I agree that it makes sense to kill the process.

When shutdown() is called explicitly, I expect an exception if it fails to stop it, or kill the process. I'm fine with killing the process.

Reply:
I added the kill()+join() calls.

Reply:
agreed on falling back to kill() after the timeout.